app/src/main/java/com/babylon/wallet/android/data/repository/TransactionStatusClient.kt

```diff
@@ -110,9 +140,31 @@ class TransactionStatusClient @Inject constructor(
     }
 }
 
+sealed interface InteractionStatusData {
+
+    val txId: String
+    val requestId: String
+}
+
 data class TransactionStatusData(
-    val txId: String,
-    val requestId: String,
+    override val txId: String,
+    override val requestId: String,
     val result: Result<Unit>,
```

Comment:
In the future I would expect to do the same as you did with the `Status` for `TransactionStatusData` too. I don't think that a `Result` is the correct type keeping here.